src/server/routes/api/v1/account.ts

```diff
@@ -0,0 +1,286 @@
+        else if (!actor.admin) return [400, "cannot promote yourself"]
+        else return [400, "cannot demote an admin"]
+    }
+}
+
+router.use(getAccount)
+router.all("/:user", async (ctx, next) => {
+    let acc = 
+        ctx.req.param("user") == "me" 
+        ? ctx.get("account") 
+        : (
+            ctx.req.param("user").startsWith("@")
+            ? Accounts.getFromUsername(ctx.req.param("user").slice(1))
+            : Accounts.getFromId(ctx.req.param("user"))
+        )
+    if (
+        acc != ctx.get("account") 
+        && !ctx.get("account")?.admin
+    ) return ServeError(ctx, 403, "you cannot manage this user")
+    if (!acc) return ServeError(ctx, 404, "account does not exist")
+
+    ctx.set("target", acc)
+
+    return next()
+})
+
+export default function (files: Files) {
+
+    router.post("/", async (ctx) => {
+        const body = await ctx.req.json()
+        if (!Configuration.accounts.registrationEnabled) {
+            return ServeError(ctx, 403, "account registration disabled")
+        }
+
+        if (auth.validate(getCookie(ctx, "auth")!)) {
+            return ServeError(ctx, 400, "you are already logged in")
+        }
+
+        if (Accounts.getFromUsername(body.username)) {
+            return ServeError(
+                ctx,
+                400,
+                "account with this username already exists"
+            )
+        }
+
+        if (body.username.length < 3 || body.username.length > 20) {
+            return ServeError(
+                ctx,
+                400,
+                "username must be over or equal to 3 characters or under or equal to 20 characters in length"
+            )
+        }
+
+        if (
+            (body.username.match(/[A-Za-z0-9_\-\.]+/) || [])[0] != body.username
+        ) {
+            return ServeError(ctx, 400, "username contains invalid characters")
+        }
+
+        if (body.password.length < 8) {
+            return ServeError(
+                ctx,
+                400,
+                "password must be 8 characters or longer"
+            )
+        }
+
+        return Accounts.create(body.username, body.password)
+            .then((account) => {
+                setCookie(ctx, "auth", auth.create(account, 3 * 24 * 60 * 60 * 1000), {
+                    path: "/",
+                    sameSite: "Strict",
+                    secure: true,
+                    httpOnly: true
+                })
+                return ctx.status(200)
+            })
+            .catch(() => {
+                return ServeError(ctx, 500, "internal server error")
+            })
+    })
+
+    router.patch(
+        "/:user",
+        requiresAccount,
+        requiresPermissions("manage"),
+        async (ctx) => {
+            let body = await ctx.req.json() as UserUpdateParameters
+            
+        }
+    )
+
+    router.patch(
+        "/:user",
+        requiresAccount,
+        requiresPermissions("manage"),
```

Comment:
```suggestion
        requiresPermissions("manage"),
```
Pass `token` to validators or something so that we can prevent apps from changing a user's password and shit 🙃